### PR TITLE
Implemented solver_workflow in solver mode

### PR DESCRIPTION
**Pull request by prmukherj**

### CODE REVIEW

File: codegen/datamodelgen.py [hunk 1]
```diff
@@ -121,6 +121,9 @@ def __init__(self):
             "preferences": DataModelStaticInfo(
                 "preferences", ("meshing", "solver", "flicing,"), self.version
             ),
+            "solver_workflow": DataModelStaticInfo(
+                "solverworkflow", ("solver",), self.version
```
Comment (prmukherj) [inline codegen/datamodelgen.py:125]: @seanpearsonuk, WE want to implement "solverworkflow" as "solver_workflow" in pyfluent?

Comment (seanpearsonuk) [reply codegen/datamodelgen.py:125]: @prmukherj no, we ought to align it with what's in the Python console, so "solverworkflow".

Comment (prmukherj) [reply codegen/datamodelgen.py:125]: @seanpearsonuk, Thank you

File: src/ansys/fluent/core/session_solver.py [hunk 2]
```diff
@@ -107,3 +113,10 @@ def preferences(self):
         if self._preferences is None:
             self._preferences = _get_preferences(self)
         return self._preferences
+
+    @property
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_solver.py:117]: @prmukherj in the Fluent solver's Python console, we have both `solverworkflow` _and_ `workflow`. `solverworkflow` is analogous to `meshing`, whcih just has the command implementations.

**Comment by seanpearsonuk** [comment]
Next steps:
- add workflow object in solver mode
- make these objects available in journaling by updating setup_for_fluent
- make the workflow extension that we did for meshing workflow available for solver workflow as well

File: src/ansys/fluent/core/workflow.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+from ansys.fluent.core.services.datamodel_se import MakeReadOnly, PyCallableStateObject
```
Comment (prmukherj) [inline src/ansys/fluent/core/workflow.py:1]: @seanpearsonuk, Is this the structure that we are expecting?

**Review by seanpearsonuk — APPROVED** [review]

**Comment by seanpearsonuk** [comment]
@ypatel-qa with this update, solverworkflow + workflow are available in PyFluent including support for Python journals.

**Review by hpohekar — APPROVED** [review]

**Pull request merged**